src/lsm/manifest_level.zig

```diff
@@ -54,6 +54,101 @@ pub fn ManifestLevelType(
             tables: std.BoundedArray(TableInfoReference, constants.lsm_growth_factor),
         };
 
+        const LevelKeyRange = struct {
+            key_min: ?Key = null,
+            key_max: ?Key = null,
+
+            fn get(range: *const LevelKeyRange) ?KeyRange {
+                assert(range.empty() or range.non_empty());
+                if (range.key_min == null and range.key_max == null) return null;
+                return KeyRange{ .key_min = range.key_min.?, .key_max = range.key_max.? };
+            }
+
+            fn update(range: *LevelKeyRange, action: union(enum) {
+                include: KeyRange,
+                exclude: KeyRange,
+            }) void {
+                switch (action) {
+                    .include => range.include(action.include),
+                    .exclude => range.exclude(action.exclude),
+                }
+            }
```

Comment:
Let's rather get rid of this magic proxy and call the functions directly. This eliminates code, and eliminates the branch.